tests/test_lnpeer.py

```diff
@@ -2121,6 +2672,106 @@ async def test_payment_trampoline_e2e_alice_t1_carol_t2_edward(self):
+        """
+
+        # store a modified trampoline onion to be injected into lnworker.new_onion_packet later when sending the htlcs
+        modified_trampoline_onion = None
+        def modified_new_onion_packet_trampoline(payment_path_pubkeys, session_key, hops_data: List[OnionHopsDataSingle], **kwargs):
+            nonlocal modified_trampoline_onion
+            assert modified_trampoline_onion is None, "this mock should get called only once"
+            modified_hops_data = copy.copy(hops_data)
+            # first payload (i[0]) is for bob who is supposed to forward the trampoline payment, in this
+            # test he should fail the incoming htlcs as their trampolines are not similar
+            new_payload = dict(modified_hops_data[0].payload)
+            amt_to_forward = dict(new_payload['amt_to_forward'])
+            amt_to_forward['amt_to_forward'] -= 1
+            new_payload['amt_to_forward'] = amt_to_forward
+            modified_hops_data[0] = dataclasses.replace(modified_hops_data[0], payload=MappingProxyType(new_payload))
+            self.logger.debug(f"{modified_hops_data=}\nsent_{hops_data=}")
+            modified_trampoline_onion = electrum.lnonion.new_onion_packet(
+                payment_path_pubkeys,
+                session_key,
+                modified_hops_data,
+                **kwargs
+            )
+            # return the unmodified onion
+            return electrum.lnonion.new_onion_packet(
+                payment_path_pubkeys,
+                session_key,
+                hops_data,
+                **kwargs
+            )
+
+        # this gets called in lnworker per sent htlc, for one sent htlc we inject the modified trampoline
+        # onion created before in the mock above
+        def modified_new_onion_packet_lnworker(payment_path_pubkeys, session_key, hops_data: List[OnionHopsDataSingle], **kwargs):
+            nonlocal modified_trampoline_onion
+            hops_data = copy.copy(hops_data)
+            if modified_trampoline_onion:
+                assert isinstance(modified_trampoline_onion, OnionPacket)
+                assert len(hops_data) == 1
+                new_payload = dict(hops_data[0].payload)
+                new_payload['trampoline_onion_packet'] = {
+                    "version": modified_trampoline_onion.version,
+                    "public_key": modified_trampoline_onion.public_key,
+                    "hops_data": modified_trampoline_onion.hops_data,
+                    "hmac": modified_trampoline_onion.hmac,
+                }
+                hops_data[0] = dataclasses.replace(hops_data[0], payload=MappingProxyType(new_payload))
+                modified_trampoline_onion = None
+            # return the unmodified onion
+            return electrum.lnonion.new_onion_packet(
+                payment_path_pubkeys,
+                session_key,
+                hops_data,
+                **kwargs
+            )
```

Comment:
the comment is a bit confusing: it returns an onion with modified `hops_data`
EDIT: well, for one HTLC.  but not for the other one.